disqus/rest-api/class-disqus-rest-api.php

```diff
@@ -738,25 +738,23 @@ private function comment_data_from_post( $post ) {
         ) );
 
         if ( $post_query->have_posts() ) {
-            $wp_post_id = $post_query->post->ID;
-            wp_reset_postdata();
+            $wp_post_id = $post_query->posts[0]->ID;
         }
 
         // If that doesn't exist, get the  and update the matching post metadata.
-        if ( null === $wp_post_id || false === $wp_post_id ) {
+        if ( empty($wp_post_id) ) {
             $identifiers = $thread['identifiers'];
             $first_identifier = count( $identifiers ) > 0 ? $identifiers[0] : null;
 
-            if ( null !== $first_identifier ) {
-                $ident_parts = explode( ' ', $first_identifier, 2 );
```

Comment:
I checked what ` $thread['identifiers']` returns and that code made no sense. That's why the synchronization wasn't working.